packages/tests/src/e2e/samples/middleWare.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { TemplateProjectFolder } from "../../utils/constants";
+import { Executor } from "../../utils/executor";
+import { expect } from "chai";
+import fs from "fs-extra";
+import path from "path";
+import { removeTeamsAppExtendToM365, editDotEnvFile } from "../commonUtils";
+import { getUuid } from "../../commonlib/utilities";
+
+export const middleWareMap: Record<
+  TemplateProjectFolder,
+  (
+    sampleName: TemplateProjectFolder,
+    testFolder: string,
+    appName: string,
+    projectPath: string,
+    steps?: {
+      create?: boolean;
+      afterCreate?: boolean;
+      beforeProvision?: boolean;
+    }
+  ) => Promise<void>
+> = {
+  [TemplateProjectFolder.HelloWorldTabBackEnd]: commonMiddleWare,
+  [TemplateProjectFolder.ContactExporter]: commonMiddleWare,
+  [TemplateProjectFolder.HelloWorldBotSSO]: commonMiddleWare,
+  [TemplateProjectFolder.TodoListSpfx]: TodoListSpfxMiddleWare,
+  [TemplateProjectFolder.MyFirstMetting]: commonMiddleWare,
+  [TemplateProjectFolder.TodoListM365]: commonMiddleWare,
+  [TemplateProjectFolder.NpmSearch]: commonMiddleWare,
+  [TemplateProjectFolder.ProactiveMessaging]: proactiveMessagingMiddleWare,
+  [TemplateProjectFolder.AdaptiveCard]: commonMiddleWare,
+  [TemplateProjectFolder.IncomingWebhook]: commonMiddleWare,
+  [TemplateProjectFolder.StockUpdate]: commonMiddleWare,
+  [TemplateProjectFolder.QueryOrg]: commonMiddleWare,
+  [TemplateProjectFolder.GraphConnector]: commonMiddleWare,
+  [TemplateProjectFolder.OneProductivityHub]: commonMiddleWare,
+  [TemplateProjectFolder.TodoListBackend]: TodoListBackendMiddleWare,
+  [TemplateProjectFolder.ShareNow]: shareNowMiddleWare,
+  [TemplateProjectFolder.Dashboard]: commonMiddleWare,
+  [TemplateProjectFolder.OutlookSignature]: outlookSignatureMiddleWare,
+  [TemplateProjectFolder.OutlookTab]: commonMiddleWare,
+  [TemplateProjectFolder.AssistDashboard]: assistantDashboardMiddleWare,
+  [TemplateProjectFolder.DiceRoller]: commonMiddleWare,
+  [TemplateProjectFolder.ChefBot]: commonMiddleWare,
+  [TemplateProjectFolder.Deeplinking]: commonMiddleWare,
+};
+
+async function proactiveMessagingMiddleWare(
```

Comment:
it's better to move these functions to the specific case better. case owner should cover its details case by case.